Game/ConsoleApplication2/MovingGameObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "GameObject.h"
+#include "Constants.h"
+
+
+class MovingGameObject : public GameObject
+{
+public:
+	// With acceleration
+	MovingGameObject(Rect bounds, Vector2D acceleration, Vector2D maxSpeed, bool isMiniMapObject = false)
+		: GameObject(bounds),
+		m_ACCELERATION(acceleration),
+		m_MAXSPEED(maxSpeed) {}
+
+	// Without acceleration
+	MovingGameObject(Rect bounds, Vector2D maxSpeed, bool isMiniMapObject = false)
+		: GameObject(bounds),
+		m_ACCELERATION(sf::Vector2f(-1, -1)),
+		m_MAXSPEED(maxSpeed),
+		m_speed(maxSpeed) {}
+
+protected:
+	const Vector2D m_ACCELERATION;
+	const Vector2D m_MAXSPEED;
+
+	Vector2D m_direction;
+	Vector2D m_speed;
+
+
+	virtual void UpdatePosition(float dt)
+	{
+		m_bounds.pos.x += (m_direction.x * (m_speed.x * dt));
+		m_bounds.pos.y += (m_direction.y * (m_speed.y * dt));
+	}
+
+	virtual void UpdateXSpeed(float dt, int multiplier, float targetSpeed)
```

Comment:
For two reasons. 
1) There will be plenty of objects that move along one axis.
2) There are situations when different axes need to be updated separately (take Player's update for example). So it's kinda necessary, unless you want to use the one function and call it twice at different times.. This means requiring x or y values to be set to 0, up to twice as many computations, and messier code.

I was debating with myself whether to have the vector update also, but thought it was unnecessary.